src/Mobs/Monster.cpp

```diff
@@ -252,6 +259,64 @@ void cMonster::MoveToWayPoint(cChunk & a_Chunk)
 
 
 
+bool cMonster::EnsureProperDestination(cChunk & a_Chunk)
+{
+	cChunk * Chunk = a_Chunk.GetNeighborChunk(m_FinalDestination.x, m_FinalDestination.z);
+	BLOCKTYPE BlockType;
+	NIBBLETYPE BlockMeta;
+	int RelX = m_FinalDestination.x - Chunk->GetPosX() * cChunkDef::Width;
+	int RelZ = m_FinalDestination.z - Chunk->GetPosZ() * cChunkDef::Width;
+	if ((Chunk != nullptr) && Chunk->IsValid())
+	{
+		// If destination in the air, go down to the lowest air block.
+		for (;;)
+		{
+			Chunk->GetBlockTypeMeta(RelX, m_FinalDestination.y - 1, RelZ, BlockType, BlockMeta);
+			if (cBlockInfo::IsSolid(BlockType))
+			{
+				break;
+			}
+			m_FinalDestination += Vector3d(0, -1, 0);
```

Comment:
I think it'd be better to use `m_FinalDestination.y += 1` instead, as that will not use a needless new object.